src/SupervisionSuccess/__tests__/SupervisionSuccess.test.js

```diff
@@ -1,100 +1,60 @@
 import React from "react";
 import { act, render } from "@testing-library/react";
 
-import params from "../model/__mocks__/params.mock";
-
 import SupervisionSuccess from "../SupervisionSuccess";
-import SupervisionSuccessComponent from "../components/SupervisionSuccess";
-import produceProjections from "../model/produceProjections";
+import LoadingScreen from "../components/LoadingScreen";
+import SupervisionSuccessContainer from "../SupervisionSuccessContainer";
+import convertCSVStringToParams from "../utils/convertCSVStringToParams";
 
-jest.mock("../model/produceProjections");
-jest.mock("../components/SupervisionSuccess", () => ({
+jest.mock("../components/LoadingScreen", () => ({
+  __esModule: true,
+  default: jest.fn().mockReturnValue(null),
+}));
+jest.mock("../components/ErrorScreen", () => ({
+  __esModule: true,
+  default: jest.fn().mockReturnValue(null),
+}));
+jest.mock("../SupervisionSuccessContainer", () => ({
   __esModule: true,
   default: jest.fn().mockReturnValue(null),
 }));
+jest.mock("../utils/convertCSVStringToParams");
 
 describe("SupervisionSuccess tests", () => {
-  const mockState = Object.keys(params)[1];
-  const mockImplementationPeriod = 9248;
-  const mockChartData = "some chart data";
-  const mockFinalPopulation = 14920;
-  const mockSavings = "some savings";
-  const mockPrisonPopulationDiff = "some diff";
-  const mockProjections = "some projections";
-  const mockChangeInRevocations = "some change in revo";
+  const mockParams = "some params";
+  const mockPath = "some path";
 
   beforeAll(() => {
-    produceProjections.mockReturnValue({
-      chartData: mockChartData,
-      savings: mockSavings,
-      prisonPopulationDiff: mockPrisonPopulationDiff,
-      finalPopulation: mockFinalPopulation,
-    });
+    jest.spyOn(window, "fetch");
   });
 
   beforeEach(() => {
     jest.clearAllMocks();
   });
 
-  it("should render SupervisionSuccessComponent", () => {
-    render(<SupervisionSuccess params={params} />);
-
-    expect(SupervisionSuccessComponent).toBeCalled();
-  });
-
-  it("should be updated with transformed data on mount", () => {
-    render(<SupervisionSuccess params={params} />);
-
-    expect(produceProjections).toBeCalled();
-    expect(SupervisionSuccessComponent.mock.calls[1][0]).toMatchObject({
-      savings: mockSavings,
-      chartData: mockChartData,
-      prisonPopulationDiff: mockPrisonPopulationDiff,
-      finalPopulation: mockFinalPopulation,
-    });
-  });
-
-  it("should change state", () => {
-    render(<SupervisionSuccess params={params} />);
+  it("should render LoadingScreen while loading", () => {
```

Comment:
It looks like we have a test case for showing the loading screen and for properly mounting once loaded, but can we also add a test case for showing the _error screen_ when an error occurs while processing the CSV?

Reply:
There is error screen test below